rapids-4-spark-tools/src/test/scala/com/nvidia/spark/rapids/tool/profiling/QualificationInfoUtils.scala

```diff
@@ -0,0 +1,182 @@
+
+import org.apache.spark.internal.Logging
+import org.apache.spark.rapids.TestUtils
+import org.apache.spark.sql.{Dataset, SparkSession}
+import org.apache.spark.sql.functions.{col, udf}
+
+// class used for testing
+case class RapidsFriends(name: String, friend: String, age: Int)
+
+/**
+ * Utilities to generate event logs used for qualification testing.
+ */
+object QualificationInfoUtils extends Logging {
+
+  def randomStringFromCharList(length: Int, chars: Seq[Char]): String = {
+    val sb = new StringBuilder
+    for (i <- 1 to length) {
+      val randomNum = util.Random.nextInt(chars.length)
+      sb.append(chars(randomNum))
+    }
+    sb.toString
+  }
+
+  def randomAlpha(length: Int): String = {
+    val chars = ('a' to 'z') ++ ('A' to 'Z')
+    randomStringFromCharList(length, chars)
+  }
+
+  def randomString(length: Int) = {
+    val r = new scala.util.Random
+    val sb = new StringBuilder
+    for (i <- 1 to length) {
+      sb.append(r.nextPrintableChar)
+    }
+    sb.toString
+  }
+
+  val randForInt = new scala.util.Random(11)
+  def randomInt(): Int = {
+    randForInt.nextInt(100)
+  }
+
+  def generateFriendsDataset(spark: SparkSession): Dataset[RapidsFriends] = {
+    import spark.implicits._
+    val df = spark.sparkContext.parallelize(
+      Seq.fill(1000){(randomAlpha(10), randomAlpha(5), randomInt)})
+      .toDF("name", "friend", "age")
+    df.as[RapidsFriends]
+  }
+
+  // dataset operations in plan show up as Lambda
+  def genDatasetEventLog(spark: SparkSession, size: Int = 1000) = {
+    import spark.implicits._
+    TestUtils.withTempPath { jsonOutFile =>
+      val ds = generateFriendsDataset(spark)
+      val dsAge = ds.filter(d => d.age > 25).map(d => (d.friend, d.age))
+      dsAge.write.json(jsonOutFile.getCanonicalPath)
+    }
+  }
+
+  def parseAge = (age: Int) => {
+    val majorAge = Seq("21", "25", "55", "18")
+    if (majorAge.contains(age)) {
+      "MILESTONE"
+    }
+    else {
+      "other"
+    }
+  }
+
+  // UDF with dataset, shows up with Lambda
+  def genUDFDSEventLog(spark: SparkSession, size: Int = 1000) = {
+    import spark.implicits._
+    TestUtils.withTempPath { jsonOutFile =>
+      val ageFunc = udf(parseAge)
+      val ds = generateFriendsDataset(spark)
+      ds.withColumn("ageCategory",ageFunc(col("age")))
+      val dsAge = ds.filter(d => d.age > 25).map(d => (d.friend, d.age))
+      dsAge.write.json(jsonOutFile.getCanonicalPath)
+    }
+  }
+
+  def cleanCountry = (country: String) => {
+    val allUSA = Seq("US", "USa", "USA", "United states", "United states of America")
+    if (allUSA.contains(country)) {
+      "USA"
+    }
+    else {
+      "unknown"
+    }
+  }
+
+  // if register UDF with udf function it shows up in plan with UDF
+  // Registering udf like:
+  //   val normaliseCountry = spark.udf.register("normalisedCountry",cleanCountry)
+  // doesn't seem to put anything unique in the plan
+  def genUDFFuncEventLog(spark: SparkSession, size: Int = 1000) = {
```

Comment:
Nit: I am not sure if we need a default value for `size=1000` here. We are setting size=1000 in the main if it is not passed in command line.

Reply:
It makes sense. Thanks!